include/clap/factory/draft/preset-discovery.h

```diff
@@ -0,0 +1,315 @@
+      const struct clap_preset_discovery_metadata_receiver *receiver);
+
+   // If the file being parsed is a preset container file (mark_as_container_file has been called)
+   // then this must be called for every preset in the file and before any preset metadata is
+   // sent with the calls below. If the file is not a container file then this should not be
+   // called at all.
+   //
+   // The path defines a human friendly path to the preset in the container file. It
+   // should be unique within the container file.
+   //
+   // The load_uri is a machine friendly string used to load the preset inside the container via a
+   // the preset-load plug-in extension. The load_uri can also just be the path if that's what the
+   // extension wants but it could also be some other unique id like a database primary key or a
+   // binary offset. It's use is entirely up to the plug-in.
+   void(CLAP_ABI *begin_contained_preset)(
+      const struct clap_preset_discovery_metadata_receiver *receiver,
+      const char                                           *path,
+      const char                                           *load_uri);
+
+   // Adds a plug-in id that this preset can be used with.
+   // plugin_abi: 0 for CLAP
+   void(CLAP_ABI *add_plugin_id)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                 uint32_t                                              plugin_abi,
+                                 const char                                           *plugin_id);
+
+   // Sets the collection to which the preset belongs to.
+   void(CLAP_ABI *set_collection_id)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                     const char *collection_id);
+
+   // Sets the flags, see clap_preset_discovery_flags.
+   // If unset, they are then inherited from the location.
+   void(CLAP_ABI *set_flags)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                             uint32_t                                              flags);
+
+   // Sets the preset name.
+   // The preset name can be infered from the file name, but it is mandatory to set it for
+   // container's preset.
+   void(CLAP_ABI *set_name)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                            const char                                           *name);
+
+   // Adds a creator name for the preset.
+   void(CLAP_ABI *add_creator)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                               const char                                           *creator);
+
+   // Sets a description of the preset.
+   void(CLAP_ABI *set_description)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                   const char *description);
+
+   // Sets the creation time and last modification time of the preset.
+   // The timestamps are in seconds since EPOCH.
```

Comment:
Maybe just explicitly say 'since the Unix epoch', since there technically are multiple reference epochs used in computing. Also, even though this is unlikely to result in issues in practice, it might be a good idea to specify whether this should be in UTC or in the user's locale timezone. Otherwise plugins/hosts will mix this up, and different platforms have different conventions.

Reply:
I've added a reference to C's `time()` function, so it is the number of seconds since EPOCH which is always UTC right?

Reply:
It is yeah. Probably doesn't hurt to specifically mention it, but I think most APIs for getting file modification times are relative to UTC. I checked Windows' filetime API and that also uses UTC, albeit with a different epoch compared to everything else.